src/masternodes/mn_checks.cpp

```diff
@@ -1715,6 +1801,492 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
+        }
+
+        // set loan scheme to default if non provided
+        if(obj.schemeId.empty()){
+            if (auto defaultScheme = mnview.GetDefaultLoanScheme()){
+                vault.schemeId = *defaultScheme;
+            } else {
+                return Res::Err("There is not default loan scheme");
+            }
+        }
+
+        // loan scheme exists
+        if (!mnview.GetLoanScheme(vault.schemeId)) {
+            return Res::Err("Cannot find existing loan scheme with id %s", vault.schemeId);
+        }
+
+        // check loan scheme is not to be destroyed
+        if (auto height = mnview.GetDestroyLoanScheme(obj.schemeId)) {
+            return Res::Err("Cannot set %s as loan scheme, set to be destroyed on block %d", obj.schemeId, *height);
+        }
+
+        auto vaultId = tx.GetHash();
+        return mnview.StoreVault(vaultId, vault);
+    }
+
+    Res operator()(const CUpdateVaultMessage& obj) const {
+
+        // vault exists
+        auto vault = mnview.GetVault(obj.vaultId);
+        if (!vault)
+            return Res::Err("Cannot find existing vault with id %s", obj.vaultId.GetHex());
+
+        // vault under liquidation
+        if(vault.val->isUnderLiquidation)
+            return Res::Err("Cannot update vault under liquidation");
+
+        // owner auth
+        if (!HasAuth(vault.val->ownerAddress)) {
+            return Res::Err("tx must have at least one input from token owner");
+        }
+
+        // loan scheme exists
+        if (!mnview.GetLoanScheme(obj.schemeId))
+            return Res::Err("Cannot find existing loan scheme with id %s", obj.schemeId);
+
+        // loan scheme is not set to be destroyed
+        if (auto height = mnview.GetDestroyLoanScheme(obj.schemeId)) {
+            return Res::Err("Cannot set %s as loan scheme, set to be destroyed on block %d", obj.schemeId, *height);
+        }
+
+        vault.val->schemeId = obj.schemeId;
+        vault.val->ownerAddress = obj.ownerAddress;
+        return mnview.StoreVault(obj.vaultId, *vault.val);
+    }
+
+
+    Res operator()(const CDepositToVaultMessage& obj) const {
+        // owner auth
+        if (!HasAuth(obj.from)) {
+            return Res::Err("tx must have at least one input from token owner");
+        }
+        // vault exists
+        auto vault = mnview.GetVault(obj.vaultId);
+        if (!vault)
+            return Res::Err("Cannot find existing vault with id %s", obj.vaultId.GetHex());
+
+        // vault under liquidation
+        if(vault.val->isUnderLiquidation)
+            return Res::Err("Cannot deposit to vault under liquidation");
+
+        //check balance
+        CalculateOwnerRewards(obj.from);
+        auto resSub = mnview.SubBalance(obj.from, obj.amount);
+        if (!resSub)
+            return Res::Err("Insufficient funds: can't subtract balance of %s: %s\n", obj.from.GetHex(), resSub.msg);
```

Comment:
@hidiego  Put here `ScriptToString(obj.from)` instead of `obj.from.GetHex()` for easier understanding of error message. This way it displays pubkey and not address.